### PR TITLE
Spelling corrections in C-Sharp compiler

### PR DESCRIPTION
**Pull request by bkoelman**
Another batch of spelling corrections in code comments and strings.

@pharring @cston @VSadov @dpoeschl would you be so kind to review these (or ask other team members)?

### CODE REVIEW

File: src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs [hunk 1]
```diff
@@ -886,7 +886,7 @@ private bool EmitFieldLoadReceiverAddress(BoundExpression receiver)
 
                 if (!field.IsStatic && EmitFieldLoadReceiverAddress(fieldAccess.ReceiverOpt))
                 {
-                    Debug.Assert(!field.IsVolatile, "volatile valuetype fields are unexpected");
+                    Debug.Assert(!field.IsVolatile, "volatile value type fields are unexpected");
```
Comment (cston) [inline src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs]: Should be 'valuetype' or 'ValueType'.

File: src/Compilers/CSharp/Portable/CodeGen/EmitOperators.cs [hunk 2]
```diff
@@ -179,7 +179,7 @@ private void EmitShortCircuitingOperator(BoundBinaryOperator condition, bool sen
             EmitCondBranch(condition.Left, ref lazyFallThrough, stopSense);
             EmitCondExpr(condition.Right, sense);
 
-            // if fallthrough was not initialized, no one is going to take that branch
+            // if fall-through was not initialized, no one is going to take that branch
```
Comment (pharring) [inline src/Compilers/CSharp/Portable/CodeGen/EmitOperators.cs]: no-one

File: src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs (hunk 1 above)
Comment (bkoelman) [inline src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs]: I'll change that. Which do you prefer?
Op 14 jul. 2015 19:52 schreef "Charles Stoner" <notifications@github.com>:

File: src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs [hunk 3]
```diff
@@ -187,7 +187,7 @@ internal static class ParameterHelpers
             // reference type, type parameter type, and so on.
             // We should consider simply allowing this in the spec.
             //
-            // Also when valuetype S has a parameterless constructor, 
+            // Also when value type S has a parameterless constructor,
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs]: valuetype or ValueType

**Comment by cston** [comment]
LGTM

**Comment by pharring** [comment]
:+1:

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs (hunk 1 above)
Comment (VSadov) [inline src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs]: I think either way is ok. It is just an assert text and either way it is sufficiently clear what went wrong.

**Comment by VSadov** [comment]
@dotnet-bot retest this please

File: src/Compilers/CSharp/Portable/CodeGen/EmitOperators.cs (hunk 2 above)
Comment (bkoelman) [inline src/Compilers/CSharp/Portable/CodeGen/EmitOperators.cs]: Nice! I have updated similar cases.

**Comment by bkoelman** [comment]
Thanks for review.

**Pull request merged**